### PR TITLE
calculate cost for exit tiles

### PR DESCRIPTION
**Pull request by Waclaw-I**
https://user-images.githubusercontent.com/14911927/153449535-d730c3c2-fedb-4fdb-a5c1-c30c3a40ad2f.mp4

### CODE REVIEW

File: front/src/Utils/PathfindingManager.ts [hunk 1]
```diff
@@ -8,19 +8,26 @@ export class PathfindingManager {
...
+        scene: Phaser.Scene,
+        collisionsGrid: number[][],
+        walkingCostGrid: number[][],
+        tileDimensions: { width: number; height: number }
+    ) {
         this.scene = scene;
 
         this.easyStar = new EasyStar.js();
-        this.easyStar.enableDiagonals();
+        // this.easyStar.enableDiagonals();
```
Comment (moufmouf) [inline front/src/Utils/PathfindingManager.ts:20]: Out of curiosity, why did you need to disable diagonals?

1 | 1
-|-
1|1000

I'm assuming in this case, if you go from top right to bottom left, the cost will be small, but you risk walking on the bottom right tile?

Comment (Waclaw-I) [reply front/src/Utils/PathfindingManager.ts:20]: Currently we will get teleported if we barely touch these fields, so diagonal movement is not an option sadly :(

**Comment by moufmouf** [comment]
Hey @Waclaw-I ,

I gave a bit of thought about this.
I'd really like to be able to keep the diagonal movement. It makes the movement seems more "natural".
Otherwise, the up-left-up-left-up-left movement is a bit weird when you walk in diagonal.

I think I came up with another idea.

We could compute the collision grid dynamically, at the moment the mouse is clicked.

Something like:

this.gameMap.getCollisionGrid(x, y)

By default, exit tiles are collidable, unless...
the (x, y) tile is part of an exit layer. In this case, the whole layer is considered as non collidable.

If we compute the collision grid at the moment of the click, we don't have to worry about updating the collision grid when the map is edited.

By the way, I just realized that if we are using the `WA.room.setTiles` (https://workadventu.re/map-building/api-room.md#changing-tiles), the "collides" property of the new tile will not be accounted for. But if we compute the collidable grid on click, we don't have such a problem. The only question is whether the performance of creating this array will be acceptable for the user experience.
